cpp/src/arrow/visitor_inline.h

```diff
@@ -17,8 +17,9 @@
 
 // Private header, not to be exported
```

Comment:
Also not for this PR:
I haven't noticed this comment before. This seems inaccurate, since we do install it (its name doesn't contain `_internal`)